power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/iterative_linear_se_solver.hpp

```diff
@@ -638,7 +638,7 @@ class IterativeLinearSESolver {
 
         // loop to iterate
         Idx num_iter = 0;
-        while (max_dev > err_tol) {
+        while (max_dev > err_tol || (err_tol == std::numeric_limits<double>::infinity() && num_iter == 0)) {
```

Comment:
Use `std::isinf` intead of `==`

Reply:
i knew there was one but i couldn't find it 😄 TY

Reply:
Why do you use `do` - `while` loop in another refactor, but not here?

Reply:
> Why do you use `do` - `while` loop in another refactor, but not here?

whoops i forgot about this one